examples/twitter-compose-with-typeahead/src/Autocomplete.tsx

```diff
@@ -0,0 +1,287 @@
+import { AutocompleteOptions } from '@algolia/autocomplete-core';
+import {
+  getAlgoliaResults,
+  parseAlgoliaHitHighlight,
+} from '@algolia/autocomplete-preset-algolia';
+import type { Hit } from '@algolia/client-search';
+import algoliasearch from 'algoliasearch/lite';
+import React, { Fragment, useRef } from 'react';
+import getCaretCoordinates from 'textarea-caret';
+
+import { useAutocomplete } from './hooks';
+import type { Account, AutocompleteItem } from './types';
+import { isValidTwitterUsername, replaceAt } from './utils';
+
+const searchClient = algoliasearch(
+  'latency',
+  'a4390aa69f26de2fd0c4209ff113a4f9'
+);
+
+export function Autocomplete(
+  props: Partial<AutocompleteOptions<AutocompleteItem>>
+) {
+  const inputRef = useRef<HTMLTextAreaElement>(null);
+  const { autocomplete, state } = useAutocomplete({
+    ...props,
+    id: 'twitter-autocomplete',
+    getSources({ query }) {
+      const cursorPosition = inputRef.current?.selectionEnd;
+      const tokenizedQuery = query
+        .split(/[\s\n]/)
+        .reduce((acc, word, index) => {
+          const previous = acc[index - 1];
+          const start = index === 0 ? index : previous.range[1] + 1;
+          const end = start + word.length;
+
+          return acc.concat([{ word, range: [start, end] }]);
+        }, [] as Array<{ word: string; range: [number, number] }>);
+
+      if (cursorPosition) {
+        const activeToken = tokenizedQuery.find((token) =>
+          token.range.includes(cursorPosition)
+        );
+
+        if (activeToken?.word && isValidTwitterUsername(activeToken?.word)) {
+          return [
+            {
+              sourceId: 'accounts',
+              onSelect({ item, setQuery }) {
+                const [index] = activeToken.range;
+                const replacement = `@${item.handle}`;
+                const newQuery = replaceAt(query, replacement, index);
+
+                setQuery(newQuery);
+              },
+              getItems() {
+                return getAlgoliaResults({
+                  searchClient,
+                  queries: [
+                    {
+                      indexName: 'autocomplete_twitter_accounts',
+                      query: activeToken.word.slice(1),
+                      params: {
+                        hitsPerPage: 8,
+                      },
+                    },
+                  ],
+                });
+              },
+            },
+          ];
+        }
+      }
+
+      return [];
+    },
+  });
+
+  const { top, height } = inputRef.current
+    ? getCaretCoordinates(inputRef.current, inputRef.current?.selectionEnd)
+    : { top: 0, height: 0 };
+
+  return (
+    <div {...autocomplete.getRootProps({})}>
+      <div className="box">
+        <div className="box-body">
+          <div className="box-avatar">
+            <img src="https://robohash.org/autocomplete" alt="" />
```

Comment:
That's a bit better?

```suggestion
            <img src="https://robohash.org/autocomplete" alt="You" />
```

Reply:
Yes for this one there's no other hint so indeed it's not decorative 👍